packages/shopping/src/application.ts

```diff
@@ -127,4 +145,41 @@ export class ShoppingApplication extends BootMixin(
 
     this.bind(UserServiceBindings.USER_SERVICE).toClass(MyUserService);
   }
+
+  async migrateSchema(options?: SchemaMigrationOptions) {
+    await super.migrateSchema(options);
+
+    const productRepo = await this.getRepository(ProductRepository);
+    await productRepo.deleteAll();
+    const productsDir = path.join(__dirname, '../fixtures/products');
+    const productFiles = fs.readdirSync(productsDir);
+
+    for (const file of productFiles) {
+      if (file.endsWith('.yml')) {
+        const productFile = path.join(productsDir, file);
+        const product = YAML.load(productFile);
+        await productRepo.create(product);
+      }
+    }
+
+    const passwordHasher = await this.get(
+      PasswordHasherBindings.PASSWORD_HASHER,
+    );
+    const userRepo = await this.getRepository(UserRepository);
+    await userRepo.deleteAll();
+    const usersDir = path.join(__dirname, '../fixtures/users');
+    const userFiles = fs.readdirSync(usersDir);
+
+    for (const file of userFiles) {
+      if (file.endsWith('.yml')) {
+        const userFile = path.join(usersDir, file);
+        const input = new NewUser(YAML.load(userFile));
+        const password = await passwordHasher.hashPassword(input.password);
```

Comment:
I am seeing this code pattern duplicated in quite few places now - in our controllers, acceptance tests and now in the migration script too.

IMO, it's time to enhance our user service and add a function for adding new users, one that can be used from all places. It's out of scope of this pull request, but please consider adding this task on your checklist.

Reply:
Follow up task https://github.com/strongloop/loopback4-example-shopping/issues/480.